internals/secrethub/client_factory.go

```diff
@@ -88,6 +89,18 @@ func (f *clientFactory) NewClientWithCredentials(provider credentials.Provider)
 	return client, nil
 }
 
+func (f *clientFactory) NewClientWithSetupCode(code string) (secrethub.ClientInterface, error) {
+	options := f.baseClientOptions()
+	options = append(options, secrethub.WithSetupCode(code))
+
+	client, err := secrethub.NewClient(options...)
+	if err != nil {
+		return nil, err
+	}
+
+	return client, nil
+}
```

Comment:
Shouldn't we just make `credentials.UseSetupCode` a credential? Then we do not have to add this function. Feels a bit superfluous because it is practically the same as above.

Reply:
Simon and I thought that it makes more sense to not have it as a credential because it is only used for authentication, and not for decryption. Implying that it is a credential could be confusing for this reason. But I agree, in terms of code duplication it is not a good idea. What do you think?

Reply:
I think that if we make sure that the client can handle this situation correctly (so no nil dereference panics) and it is documented, there should be no real problem. Using anything from the `credentials` package requires some documentation anyway. For example, you cannot use `credentials.UseAWS()` without knowing that this requires an AWS role to be assumed.